src/core/include/openvino/core/type.hpp

```diff
@@ -37,8 +37,8 @@ struct OPENVINO_API DiscreteTypeInfo {
     // exact type identification
     const DiscreteTypeInfo* parent;
 
-    DiscreteTypeInfo() = default;
     OPENVINO_SUPPRESS_DEPRECATED_START
+    DiscreteTypeInfo() = default;
```

Comment:
I hope we don't deprecate default constructor

Reply:
all default implementation must be located under `OPENVINO_SUPPRESS_DEPRECATED_START`